### ISSUE TITLE
Site creation (/start): flow stuck on previous flow (eg: launch-site)

### ISSUE THREAD

**Issue by ciampo** [Automattic/wp-calypso#50273]
**NOTE:** currently WIP in https://github.com/Automattic/wp-calypso/issues/53254 by resetting the store between flows.

___

The `/start` site creation flow may get stuck if the user is trying to create multiple sites in one session.

Not entirely sure how to consistently reproduce this behaviour (see step 2 below), but it's already happened to me a few times — simulating a slow connection may help

### Steps to (tentatively) reproduce the behaviour

1. Create a site through `/start`
2. Sometimes the flow takes a long time to "create" the site — I feel like I could have opened the `/home` page in another tab, worked on other stuff, and later come back to the original `/start` tab where then flow would have completed successfully
3. Create a new site through `/start` — either by clicking links in the UI, or by typing `wordpress.com/start` directly in the URL
4. Notice how the flow is stuck on the previous site creation, and doesn't allow the user to create a new site

#### What I expected to happen

The flow should start from the beginning

#### Screenshot / Video: If applicable, add screenshots to help explain your problem.
* Demo when doing manual testing:
https://user-images.githubusercontent.com/1083581/108513605-203d2780-72c3-11eb-87d5-8c11da5dc947.mp4

* Demo from e2e failing tests:
https://user-images.githubusercontent.com/6549265/108547595-8adc6c00-729f-11eb-8eac-9126e42b7792.mov

**Comment by worldomonation** [comment]
I've also reproduced this during both e2e tests.

Video of the local e2e test run (`Launch when having multiple sites`):
https://user-images.githubusercontent.com/6549265/108547595-8adc6c00-729f-11eb-8eac-9126e42b7792.mov

This issue has been responsible for failing e2e runs:
- [example 1](https://app.circleci.com/pipelines/github/Automattic/wp-e2e-tests-for-branches/58542/workflows/335873eb-fd0b-45df-a0df-e6c359d8549e/jobs/142691/parallel-runs/1)
- [example 2](https://app.circleci.com/pipelines/github/Automattic/wp-e2e-tests-for-branches/58541/workflows/28f9e57c-bef0-430a-ae06-a8f58f8d4448/jobs/142692)

Reproducibility appears to be intermittent in CircleCI (eg as of 2021-02-19 19:00Z it does not appear to be an issue) but I am able to reproduce consistently when running the spec locally (at any hour of the day).

**Comment by razvanpapadopol** [comment]
First time when we experienced something like this it was only affecting Launch flow (reported in https://github.com/Automattic/wp-calypso/issues/42515) and it was fixed in https://github.com/Automattic/wp-calypso/pull/42516. That was also the time when I've added `Launch when having multiple sites` e2e test.

Now this time it may be something related to the signup framework in general since the [video](https://user-images.githubusercontent.com/6549265/108547595-8adc6c00-729f-11eb-8eac-9126e42b7792.mov) @worldomonation shared indicates the flow as being stuck in [site creation](https://cloudup.com/c2hQzgMS4Ao) and not launch. If we know when those tests started to fail for the first time, we can try to narrow down to a list of potential PRs that could've introduced this regression.

Also pinging @yansern here since I remember he reproduced this issue recently in production.

**Comment by ramonjd** [comment]
I wonder if this recent [addition to the signup controller](https://github.com/Automattic/wp-calypso/pull/49680/files#diff-9d7d6e5b810ec4d930d0dd4ab43ab0a72e8082274e818ed6888ec938a94dacc2R311) has anything to do with it

It causes an error on `/start` since the `siteSlug` isn't available to be passed to the Promise.

<img width="765" alt="Screen Shot 2021-02-25 at 8 23 01 pm" src="https://user-images.githubusercontent.com/6458278/109137616-7de2df80-77ad-11eb-8166-eb853b33b35e.png">

I've created a fix just in case

https://github.com/Automattic/wp-calypso/pull/50478

🤞

**Comment by ciampo** [comment]
cc @Automattic/martech to have some eyes more familiar with `/start` signup framework on this one.

**Update**: MarTech is taking over this one p1614692244021500/1614677130.020900-slack-C012K365WR3

**Comment by ddc22** [comment]
![](https://github.trello.services/images/mini-trello-icon.png) [Site creation (/start): flow stuck on previous site launch](https://trello.com/c/SjLOY37z/19-site-creation-start-flow-stuck-on-previous-site-launch)

**Comment by ciampo** [comment]
Hey @jdc91 , is there any progress on this issue? I can't access the trello board.

**Comment by ddc22** [comment]
Hey @ciampo I actually tried to reproduce this before but couldn't, I am currently blocked by some other work priorities. @southp will take over this issue from here onwards and update you on the progress!

**Comment by ciampo** [comment]
Thank you for the update! I know it may be a bit flaky to reproduce, let me know if you can't at all!

**Comment by razvanpapadopol** [comment]
Let's keep this on hold since it may be fixed by resetting the store when starting a different flow in https://github.com/Automattic/wp-calypso/pull/52434 or a follow-up.

cc: @StefanNieuwenhuis

**Comment by StefanNieuwenhuis** [comment]
> Let's keep this on hold since it may be fixed by resetting the store when starting a different flow in #52434 or a follow-up.
> 
> cc: @StefanNieuwenhuis

Thanks for the heads-up, @razvanpapadopol. Good idea and let's keep a close eye on the store resetting issue.

**Comment by dpasque** [comment]
In retesting this, this appears to now be fixed (could be from linked PR, or the reworking and redesign of the `/start` flow)
